libraries/botbuilder-dialogs-declarative/src/resources/resource.ts

```diff
@@ -0,0 +1,17 @@
+import { debug } from "util";
+
+/**
+ * @module botbuilder-dialogs-declarative
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const path = require('path');
+const fs = require('fs');
+
+export interface IResource {
+    id(): string;
+    readText() : Promise<string>;
+}
```

Comment:
C# interface looks like this:
  public interface IResource
    {
        /// <summary>
        /// Resource name
        /// </summary>
        string Id { get; }

        /// <summary>
        /// Get resource as text async
        /// </summary>
        /// <returns></returns>
        Task<string> ReadTextAsync();

        /// <summary>
        /// Get resource as text
        /// </summary>
        /// <returns></returns>
        string ReadText();

        /// <summary>
        /// Get readonly stream
        /// </summary>
        /// <returns></returns>
        Stream OpenStream();
    }
???